service/host_helpers.go

```diff
@@ -0,0 +1,73 @@
+package service
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/model/user"
+	"github.com/evergreen-ci/evergreen/rest"
+	"github.com/evergreen-ci/evergreen/units"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/pkg/errors"
+)
+
+const (
+	InvalidStatusError             = "'%v' is not a valid status"
+	DecommissionStaticHostError    = "Cannot decommission static hosts"
+	HostTerminationQueueingError   = "Error starting background job for host termination"
+	HostUpdateError                = "Error updating host"
+	HostTerminationQueueingSuccess = "Host %v successfully queued for termination"
+	HostStatusUpdateSuccess        = "Host status successfully updated from '%v' to '%v'"
+	HostStatusWriteConfirm         = "Successfully updated host status"
+	UnrecognizedAction             = "Unrecognized action: %v"
+)
+
+func modifyHostStatus(h *host.Host, opts *uiParams, u *user.DBUser) (string, *rest.APIError) {
+	env := evergreen.GetEnvironment()
+	queue := env.RemoteQueue()
```

Comment:
I think it would be better to accept the queue as the first parameter to the function, and verify that the provided queue has been started. (if it hasn't, return error 500)